js/ace.js

```diff
@@ -0,0 +1,1021 @@
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const response = await this.publicGetOapiV2ListTradePrice (params);
+        const marketId = market['id'];
+        const ticker = this.safeValue (response, marketId, {});
+        //
+        //     {
+        //         "BTC/USDT":{
+        //             "base_volume":229196.34035399999,
+        //             "last_price":11881.06,
+        //             "quote_volume":19.2909
+        //         }
+        //     }
+        //
+        ticker['id'] = marketId;
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name ace#fetchTickers
+         * @description fetches price tickers for multiple markets, statistical calculations with the information calculated over the past 24 hours each market
+         * @param {[string]|undefined} symbols unified symbols of the markets to fetch the ticker for, all market tickers are returned if not assigned
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} an array of [ticker structures]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const response = await this.publicGetOapiV2ListTradePrice ();
+        //
+        //     {
+        //         "BTC/USDT":{
+        //             "base_volume":229196.34035399999,
+        //             "last_price":11881.06,
+        //             "quote_volume":19.2909
+        //         }
+        //     }
+        //
+        const tickers = [];
+        const pairs = Object.keys (response);
+        for (let i = 0; i < pairs.length; i++) {
+            const marketId = pairs[i];
+            const ticker = this.safeValue (response, marketId);
+            ticker['id'] = marketId;
+            tickers.push (ticker);
+        }
+        return this.parseTickers (tickers, symbols);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name ace#fetchOrderBook
+         * @description fetches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int|undefined} limit the maximum amount of order book entries to return
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'quoteCurrencyId': market['quoteId'],
+            'baseCurrencyId': market['baseId'],
+        };
```

Comment:
can you check what is going on with `fetchOrderBook` ?
```
ace.fetchOrderBook (BTC/USDT)
BadRequest ace {"attachment":null,"message":"參數錯誤","parameters":null,"status":9996}
---------------------------------------------------
[BadRequest] ace {"attachment":null,"message":"參數錯誤","parameters":null,"status":9996}

    at throwExactlyMatchedException  js/base/Exchange.js:2275              throw new exact[string] (message);                                              
    at handleErrors                  js/ace.js:1017                        this.throwExactlyMatchedException (this.exceptions['exact'], status, feedback); 
    at                               js/base/Exchange.js:598               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                  
    at async timeout                 js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                        js/base/Exchange.js:1919              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                       js/base/Exchange.js:1923              return await this.fetch2 (path, api, method, params, headers, body, config, con…
    at fetchOrderBook                js/ace.js:353                         const response = await this.publicGetOpenV2PublicGetOrderBook (this.extend (req…
    at async run                     examples/js/cli.js:281                const result = await exchange[methodName] (... args)                            

ace {"attachment":null,"message":"參數錯誤","parameters":null,"status":9996}
```

Reply:
@carlosmiei interesting, they return different baseCurrencyId in usdt market, and it didn't work. I'll ask them about this issue.